### PR TITLE
go/proxyd: Bring back method mappings

### PR DESCRIPTION
**Pull request by mslipper**
This PR brings back the ability to selectively route RPCs to different backends.

### CODE REVIEW

File: go/proxyd/server.go [hunk 1]
```diff
@@ -83,7 +92,17 @@ func (s *Server) HandleRPC(w http.ResponseWriter, r *http.Request) {
...
+  if group == "" {
+      // use unknown below to prevent DOS vector that fills up memory
+      // with arbitrary method names.
+      log.Info("blocked request for non-whitelisted method", "source", "ws", "method", req.Method)
+      RecordRPCError(ctx, BackendProxyd, MethodUnknown, ErrMethodNotWhitelisted)
+      writeRPCError(w, req.ID, ErrMethodNotWhitelisted)
+      return
+  }
+
+	backendRes, err := s.backendGroups[group].Forward(ctx, req)
```
Comment (tynes) [inline go/proxyd/server.go:105]: Perhaps program defensively here by first checking to make sure `group` is in `s.backendGroups` before calling `Forward`?

**Review by tynes — APPROVED** [review]
Generally looks good to me, assuming that you have been testing these changes locally

**Review by optimisticben — APPROVED** [review]
lgtm

File: go/proxyd/server.go (hunk 1 above)
Comment (mslipper) [reply go/proxyd/server.go:105]: You can't define an invalid method mapping in the config like that - the daemon will refuse to start.

**Pull request merged**